### PR TITLE
add Create container registry repo API

### PR DESCRIPTION
**Pull request by NishanthSpShetty**
# Description

Create repository in ECR.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# why

needed when using private repo in ECR to push images.

# How Has This Been Tested?

manual test

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pkg/service/aws/ecr.go [hunk 1]
```diff
@@ -5,24 +5,49 @@ import (
...
 	"github.com/pkg/errors"
+	"gitlab.com/netbook-devs/spawner-service/pkg/service/labels"
 	proto "gitlab.com/netbook-devs/spawner-service/proto/netbookai/spawner"
 )
 
+func asEcrTags(label map[string]string) []*ecr.Tag {
+	tags := []*ecr.Tag{}
+	for k, v := range label {
+		key := k
+		val := v
```
Comment (sid255) [inline pkg/service/aws/ecr.go:16]: why not for `key, val := range label` ?

Comment (NishanthSpShetty) [reply pkg/service/aws/ecr.go:16]: we have to use the key, value as pointers, this is classic go gotchas. the `k` and `v` in loops are mutated and if we use pointer we end up using the wrong value.

File: pkg/service/aws/ecr.go [hunk 2]
```diff
@@ -5,24 +5,49 @@ import (
 
 	"github.com/aws/aws-sdk-go/service/ecr"
 	"github.com/pkg/errors"
+	"gitlab.com/netbook-devs/spawner-service/pkg/service/labels"
 	proto "gitlab.com/netbook-devs/spawner-service/proto/netbookai/spawner"
 )
 
+func asEcrTags(label map[string]string) []*ecr.Tag {
```
[the remark this answers is not in the document]
Comment (NishanthSpShetty) [reply pkg/service/aws/ecr.go:12]: would conflict with `labels` package. let me make ot `tags`, wdyt ?

[the remark this answers is not in the document]
Comment (sid255) [reply pkg/service/aws/ecr.go:12]: Did not get you.

[the remark this answers is not in the document]
Comment (NishanthSpShetty) [reply pkg/service/aws/ecr.go]: there is a package in spawner `service/labels`. Updating to `l`

**Review by sid255 — APPROVED** [review]
LGTM

**Pull request merged**